psydac/core/bsplines_pyccel.py

```diff
@@ -0,0 +1,1018 @@
+        It should be of the appropriate shape and dtype.
+
+    Notes
+    -----
+    Contents of 2D output arrays 'quad_x' and 'quad_w' are accessed with two
+    indices (ie,iq) where:
+      . ie is the global element index;
+      . iq is the local index of a quadrature point within the element.
+
+    """
+    ne = len(breaks) - 1
+    nq = len(quad_rule_x)
+
+    # Compute location and weight of quadrature points from basic rule
+    for ie in range(len(breaks) - 1):
+        a = breaks[ie]
+        b = breaks[ie + 1]
+
+        c0 = 0.5 * (a + b)
+        c1 = 0.5 * (b - a)
+        out[ie, :, 0] = c1 * quad_rule_x[:] + c0
+        out[ie, :, 1] = c1 * quad_rule_w[:]
+
+
+# =============================================================================
+def basis_ders_on_quad_grid_p(knots: 'float[:]', degree: int, quad_grid: 'float[:,:]', nders: int, normalization: bool,
+                            out: 'float[:,:,:,:]'):
+    """
+    Evaluate B-Splines and their derivatives on the quadrature grid.
+
+    If called with normalization='M', this uses M-splines instead of B-splines.
+
+    Parameters
+    ----------
+    knots : array_like
+        Knots sequence.
+
+    degree : int
+        Polynomial degree of B-splines.
+
+    quad_grid: ndarray
+        2D array of shape (ne, nq). Coordinates of quadrature points of
+        each element in 1D domain, which can be given by quadrature_grid()
+        or chosen arbitrarily.
+
+    nders : int
+        Maximum derivative of interest.
+
+    normalization : str
+        Set to 'B' for B-splines, and 'M' for M-splines.
+
+    out : array
+        The result will be inserted into this array.
+        It should be of the appropriate shape and dtype.
+
+    Notes
+    -----
+        Values of B-Splines and their derivatives at quadrature points in
+        each element of 1D domain. Indices are
+        . ie: global element         (0 <= ie <  ne    )
+        . il: local basis function   (0 <= il <= degree)
+        . id: derivative             (0 <= id <= nders )
+        . iq: local quadrature point (0 <= iq <  nq    )
+    """
+
+    ne, nq = quad_grid.shape
+    if normalization:
+        integrals = np.zeros(knots.shape[0] - degree - 1)
+        basis_integrals_p(knots, degree, integrals)
+
+    temp_spans = np.zeros(len(knots), dtype=int)
+    actual_index = elements_spans_p(knots, degree, temp_spans)
+    spans = temp_spans[:actual_index]
+
+    ders = np.zeros((nders + 1, degree + 1))
+
+    for ie in range(ne):
+        xx = quad_grid[ie, :]
+        span = spans[ie]
+        for iq, xq in enumerate(xx):
+            basis_funs_all_ders_p(knots, degree, xq, span, nders, False, ders)
+            if normalization:
+                ders /= integrals[span - degree:span + 1]
+            for i_der in range(nders + 1):
+                for i_basis in range(degree + 1):
+                    out[ie, i_basis, i_der, iq] = ders[i_der, i_basis]
```

Comment:
Why not using the `transpose` method instead? See [here](https://github.com/pyccel/pyccel/blob/master/doc/pyccelext/numpy.rst#transpose-like-operations).
```suggestion
            out[ie, :, :, iq] = ders.T
```

Reply:
Done in 0aa50ba